### PR TITLE
Remove strange mock setup.

### PR DESCRIPTION
**Pull request by alunduil**
The mock setup was overly complex and hid many details of what was
mocked where.  This new setup ensures that mocks stay in their
respective test suites.  This will make the type hint additions much
much simpler.

### CODE REVIEW

File: etest/docker/container.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 # etest is freely distributable under the terms of an MIT-style license.
 # See COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-import re
```
Comment (LinoBigatti) [inline etest/docker/container.py:7]: I remember taking that out ._. whoops

File: etest_test/unit_test/overlay_test.py [hunk 2]
```diff
@@ -8,36 +8,31 @@
...
 
-class BaseOverlayTest(BaseEtestTest):
+@unittest.mock.patch.object(sut, "ebuild")
+class BaseOverlayTest(unittest.TestCase):
     """Overlay test."""
 
-    mocks_mask: Set = set()
-    mocks: Set = set()
-
     def setUp(self):
```
Comment (LinoBigatti) [inline etest_test/unit_test/overlay_test.py:23]: Is this supposed to be camelCase? Im guessing its an inherited method.

File: etest_test/unit_test/tests_test.py [hunk 3]
```diff
@@ -179,26 +167,19 @@ def setUp(self):
...
-        self.tests = tests.Tests()
+        self.tests = sut.Tests()
         self.assertEqual(self.tests.ebuild_selector, [])
         self.assertEqual(len(list(self.tests)), len(self.test_calls))
         self.assertEqual(self.mocked_Test.mock_calls, self.test_calls)
 
     def test_tests_with_filter_with_version(self):
         """tests.Tests(('etest-9999.ebuild',))."""
-        self.tests = tests.Tests(("etest-9999.ebuild",))
+        self.tests = sut.Tests(("etest-9999.ebuild",))
```
Comment (LinoBigatti) [inline etest_test/unit_test/tests_test.py:183]: how would this react to ebuilds with no -9999 versions?

File: etest/build.py [hunk 4]
```diff
@@ -24,4 +24,9 @@
 @click.option("--name", type=Path, default=os.path.basename(sys.argv[0]), help="Name of the directory?")
 def main(hardened: bool, multilib: bool, systemd: bool, architecture: str, environment: List[str], name: Path) -> None:
     """Build the etest image."""
-    pass
+    click.echo(f"hardened: {hardened}")
+    click.echo(f"multilib: {multilib}")
+    click.echo(f"systemd: {systemd}")
+    click.echo(f"architecture: {architecture}")
+    click.echo(f"environment: {environment}")
+    click.echo(f"name: {name}")
```
Comment (LinoBigatti) [inline etest/build.py:32]: I hope this doesnt come to bite us in the form of a merge conflict in the future.

**Review by LinoBigatti — APPROVED** [review]
Just some general questions, but overall it seems good to merge imo. Sorry for taking this long, I had some busy days.

File: etest/docker/container.py (hunk 1 above)
Comment (alunduil) [reply etest/docker/container.py:7]: It happens.  I've made the checks requirements more strict so hopefully that will help these things being missed in the future.  Let me know if anything causes issues though.

File: etest_test/unit_test/overlay_test.py (hunk 2 above)
Comment (alunduil) [reply etest_test/unit_test/overlay_test.py:23]: It is, it's inherited from unittest.TestCase which follows the jUnit conventions including camelCase.

File: etest_test/unit_test/tests_test.py (hunk 3 above)
Comment (alunduil) [reply etest_test/unit_test/tests_test.py:183]: Good question.  That sounds like a test case we could add once I make a bit more progress on cleaning things up (or if you're feeling brave, you could try adding in the current test suite).

File: etest/build.py (hunk 4 above)
Comment (alunduil) [reply etest/build.py:32]: It will but it should be easy: delete the click.echo and keep the other changes.

**Comment by alunduil** [comment]
> Just some general questions, but overall it seems good to merge imo. Sorry for taking this long, I had some busy days.

No worries.  Life happens.  I know my reviews can sometimes take a day or two as well so as long as we're making progress within a few days I think we're doing well.

I'm hoping to finally get the typing stuff figured out tonight thanks to this little stickler but some of the types I created are true monsters in the current system so it may take me a bit longer than I'm planning.

**Pull request merged**